### PR TITLE
Adapt NetworkInformation API to new events structure

### PR DESCRIPTION
**Pull request by queengooborg**
This PR adapts the NetworkInformation API to conform to the new events structure.

### CODE REVIEW

**Review by Elchi3 — CHANGES_REQUESTED** [review]
It looks to me as if typechange/ontypechange isn't a thing anymore.
Can you confirm and remove?

**Comment by queengooborg** [comment]
Running https://mdn-bcd-collector.appspot.com/tests/api/NetworkInformation/ontypechange in BrowserStack on Chrome Android 96, it appears that it's still implemented.  However, it's not standard track and there is a discussion open for Chrome's removal, see https://crbug.com/699892.

**Comment by queengooborg** [comment]
Running https://mdn-bcd-collector.appspot.com/tests/api/NetworkInformation/ontypechange in BrowserStack on Chrome Android 96, it appears that it's still implemented.  However, it's not standard track and there is a discussion open for Chrome's removal, see https://crbug.com/699892.

File: api/NetworkInformation.json [hunk 1]
```diff
@@ -457,6 +409,56 @@
...
+        "__compat": {
+          "description": "<code>typechange</code> event",
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/NetworkInformation/typechange_event",
+          "spec_url": "https://wicg.github.io/netinfo/#dom-networkinformation-onchange",
+          "support": {
+            "chrome": {
+              "version_added": false
+            },
+            "chrome_android": {
+              "version_added": "38"
```
Comment (Elchi3) [inline api/NetworkInformation.json:422]: Didn't verify if this passes CI

```suggestion
              "version_added": "38"
              "notes": "Removal proposed in <a href=' https://crbug.com/699892'>bug 699892</a>."
```

**Comment by Elchi3** [comment]
> Running https://mdn-bcd-collector.appspot.com/tests/api/NetworkInformation/ontypechange in BrowserStack on Chrome Android 96, it appears that it's still implemented. However, it's not standard track and there is a discussion open for Chrome's removal, see https://crbug.com/699892.

Ah interesting! Can you add that bug as a note? That way, people know that it is really a dead thing.

**Review by Elchi3 — APPROVED** [review]

**Pull request merged**